### PR TITLE
Use `ubi9/openjdk-21-runtime` instead of `ubi9/openjdk-21` as the base image

### PR DESCRIPTION
**Pull request by snazy**
Supersedes #8502

### CODE REVIEW

File: tools/dockerbuild/docker/Dockerfile-server [hunk 1]
```diff
@@ -20,7 +20,7 @@
 # (See site/docs/try/configuration.md)
 #
 ###
-FROM registry.access.redhat.com/ubi9/openjdk-21:1.18-4
+FROM registry.access.redhat.com/ubi9/openjdk-21-runtime:1.18-4
```
Comment (adutra) [inline tools/dockerbuild/docker/Dockerfile-server:23]: After doing a `docker build & docker save` I observed that the total uncompressed image size went down from 682 MiB to 584 MiB. Nice!

I also validated that the `JAVA_OPTS_APPEND`,`JAVA_TOOL_OPTIONS` and `JAVA_DEBUG` env vars are still working.

**Review by adutra — APPROVED** [review]
LGTM – is this worth a mention in the changelog?

File: tools/dockerbuild/docker/Dockerfile-server (hunk 1 above)
Comment (snazy) [reply tools/dockerbuild/docker/Dockerfile-server:23]: >  I observed that the total uncompressed image size went down from 682 MiB to 584 MiB. Nice!

Not so much on for amd64 - "just" down by 40MB - but still.

**Comment by snazy** [comment]
> is this worth a mention in the changelog?

Don't think so - it's the same base image, just without the Java tools.

**Pull request merged**